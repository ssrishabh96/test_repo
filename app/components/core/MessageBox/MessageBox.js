import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import { CenteredWhiteText, MessageBoxView } from './MessageBox.styles';

class MessageBox extends React.Component {
  static propTypes = {
    message: PropTypes.string.isRequired,
    type: PropTypes.string,
    timeout: PropTypes.number,
  }

  static defaultProps = {
    type: 'default',
    timeout: 2500,
  }

  state = {
    isVisible: true,
  }

  componentWillMount() {
    const { timeout } = this.props;
    setTimeout(this.hideView, timeout);
  }

  hideView = () => {
    this.setState({
      isVisible: false,
    });
  }

  render() {
    const { message, type = 'default' } = this.props;
    const { isVisible } = this.state;
    if (isVisible) {
      return (
        <MessageBoxView type={type}>
          <CenteredWhiteText>
            {message}
          </CenteredWhiteText>
        </MessageBoxView>
      );
    }
    return <View />;
  }
}

export default MessageBox;
