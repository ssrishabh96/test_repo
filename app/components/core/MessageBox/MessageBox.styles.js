import styled from 'styled-components/native';
import colors from '../../../styles/colors';

export const MessageBoxView = styled.View`
  justifyContent: center;
  alignSelf: stretch;
  marginLeft: 10;
  marginRight: 10;
  marginTop: 10;
  height: 80;
  borderColor: transparent;
  borderBottomWidth: 6;
  backgroundColor: ${props => colorMapper(props.type).backgroundColor};
  borderBottomColor: ${props => colorMapper(props.type).borderBottomColor};
`;

export const CenteredWhiteText = styled.Text`
  color: #fff;
  fontSize: 18;
  textAlign: center;
`;


const colorMapper = (type) => {
  switch (type) {
    case 'error':
      return {
        backgroundColor: 'red',
        borderBottomColor: '#7c0002',
      };
    case 'success':
      return {
        backgroundColor: '#01bd2e',
        borderBottomColor: '#006819',
      };
    default:
      return {
        backgroundColor: colors.COPART_BLUE,
        borderBottomColor: colors.COPART_BLUE,
      };
  }
};
