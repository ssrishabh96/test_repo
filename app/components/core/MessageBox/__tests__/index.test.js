import React from 'react';
import { shallow } from 'enzyme';
import MessageBox from '../MessageBox';

describe('Testing MessageBox component', () => {
  test('Renders with success type', () => {
    const wrapper = shallow(<MessageBox
      type={'success'}
      message="Success Message"
    />);
    // console.log(wrapper.props().children.props);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.prop('type')).toEqual('success');
    expect(wrapper.props().children.props.children).toBe('Success Message');
  });
  test('Renders with error type', () => {
    const wrapper = shallow(<MessageBox
      type={'error'}
      message="Error Message"
    />);
    // console.log(wrapper.props().children.props);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.prop('type')).toEqual('error');
    expect(wrapper.props().children.props.children).toBe('Error Message');
  });
  test('Renders with default props', () => {
    const wrapper = shallow(<MessageBox message="Info Message" />);
    expect(wrapper.prop.arguments).toBeNull();
    expect(wrapper.exists('<Text>Info Message</Text>')).toBeTruthy();
  });
});
