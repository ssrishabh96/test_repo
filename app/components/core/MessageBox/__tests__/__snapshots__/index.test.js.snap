// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MessageBox component Renders with error type 1`] = `
<Styled(Component)
  type="error"
>
  <Styled(Text)>
    Error Message
  </Styled(Text)>
</Styled(Component)>
`;

exports[`Testing MessageBox component Renders with success type 1`] = `
<Styled(Component)
  type="success"
>
  <Styled(Text)>
    Success Message
  </Styled(Text)>
</Styled(Component)>
`;
